src/historyMenuElements.ts

```diff
@@ -139,27 +243,36 @@ class HistoryElement extends PopupMenu.PopupSubMenuMenuItem {
             this.menu.addMenuItem(new PopupMenu.PopupMenuItem('Unknown source.'));
         }
 
-        const previewItem = new PopupMenu.PopupBaseMenuItem({can_focus: false, reactive: false});
-        this.menu.addMenuItem(previewItem);
+        this.menu.addMenuItem(new HistoryElementSubmenuSeparator());
 
         this._setAsWallpaperItem = new PopupMenu.PopupMenuItem('Set As Wallpaper');
         this._setAsWallpaperItem.connect('activate', () => {
             this.emit('activate', null); // Fixme: not sure what the second parameter should be. null seems to work fine for now.
         });
 
-        if (index !== 0) {
-            // this.menu.addMenuItem(new PopupMenu.PopupBaseMenuItem({ can_focus: false, reactive: false })); // theme independent spacing
-            this.menu.addMenuItem(this._setAsWallpaperItem);
-        }
+        this.menu.addMenuItem(this._setAsWallpaperItem);
 
-        const copyToFavorites = new PopupMenu.PopupMenuItem('Save For Later');
+        const copyToFavoritesLabelText = 'Save For Later';
+        const copyToFavorites = new PopupMenu.PopupMenuItem(copyToFavoritesLabelText);
         copyToFavorites.connect('activate', () => {
             this._saveImage().catch(error => {
                 Logger.error(error, this);
             });
         });
+        // Disable copy option if the file was already saved.
+        this.menu.connect('open-state-changed', (_, open) => {
+            if (open && this._checkAlreadySafed()) {
```

Comment:
```suggestion
            if (open && this._checkAlreadySaved()) {
```

Reply:
Thanks, same typo was also in the commit message :facepalm: !